### PR TITLE
Upgrade to latest CE3 snapshot where Sync extends MonadCancel

### PR DESCRIPTION
**Pull request by mpilquist**

### CODE REVIEW

File: core/shared/src/main/scala/fs2/Pull.scala [hunk 1]
```diff
@@ -1118,8 +1118,10 @@ private[fs2] class PullMonadErrorInstance[F[_], O] extends MonadError[Pull[F, O,
 
 private[fs2] class PullSyncInstance[F[_], O](implicit F: Sync[F])
     extends PullMonadErrorInstance[F, O]
-    with Sync[Pull[F, O, *]] {
+    with Sync[Pull[F, O, *]]
+    with MonadCancel.Uncancelable[Pull[F, O, *], Throwable] {
   def monotonic: Pull[F, O, FiniteDuration] = Pull.eval(F.monotonic)
   def realTime: Pull[F, O, FiniteDuration] = Pull.eval(F.realTime)
   def suspend[A](hint: Sync.Type)(thunk: => A): Pull[F, O, A] = Pull.eval(F.suspend(hint)(thunk))
+  def forceR[A, B](fa: Pull[F, O, A])(fb: Pull[F, O, B]): Pull[F, O, B] = flatMap(fa)(_ => fb)
```
Comment (mpilquist) [inline core/shared/src/main/scala/fs2/Pull.scala:1126]: Not sure if this is the right implementation but until there's law checking for CE3, there's no way to check.

Comment (SystemFw) [reply core/shared/src/main/scala/fs2/Pull.scala:1126]: You need to `attempt`. `fa.attempt.flatMap(_ => fb)`

Comment (mpilquist) [reply core/shared/src/main/scala/fs2/Pull.scala]: Good catch, fixed

**Comment by mpilquist** [comment]
Waiting on https://github.com/typelevel/cats-effect/pull/1563 and https://github.com/typelevel/cats-effect/pull/1556

File: core/shared/src/main/scala/fs2/CompositeFailure.scala [hunk 2]
```diff
@@ -24,9 +24,9 @@ package fs2
 import cats.data.NonEmptyList
 
 /** Represents multiple (>1) exceptions were thrown. */
-final class CompositeFailure(
-    val head: Throwable,
-    val tail: NonEmptyList[Throwable]
+case class CompositeFailure(
```
Comment (mpilquist) [inline core/shared/src/main/scala/fs2/CompositeFailure.scala:27]: cats-effect-testkit provides an `Eq[Throwable]` which uses `.equals`

File: build.sbt [hunk 3]
```diff
@@ -112,12 +112,13 @@ lazy val core = crossProject(JVMPlatform, JSPlatform)
     libraryDependencies ++= Seq(
       "org.typelevel" %%% "cats-core" % "2.3.1",
       "org.typelevel" %%% "cats-laws" % "2.3.1" % Test,
-      "org.typelevel" %%% "cats-effect" % "3.0.0-M5",
-      "org.typelevel" %%% "cats-effect-laws" % "3.0.0-M5" % Test,
-      "org.typelevel" %%% "cats-effect-testkit" % "3.0.0-M5" % Test,
+      "org.typelevel" %%% "cats-effect" % "3.0-65-7c98c86",
```
Comment (mpilquist) [inline build.sbt:115]: We need this version to pick up (1) adjustment to MonadCancel laws (2) fix in TestInstances for an integer overflow

**Review by SystemFw — APPROVED** [review]

**Pull request merged**